modules/javafx.controls/src/main/java/javafx/scene/control/skin/TableColumnHeader.java

```diff
@@ -645,9 +646,11 @@ protected void resizeColumnToFitContent(int maxRows) {
             Region r = (Region) n;
             padding = r.snappedLeftInset() + r.snappedRightInset();
         }
-
-        TableRow<T> tableRow = new TableRow<>();
-        tableRow.updateTableView(tv);
+        Callback<TableView<T>, TableRow<T>> rowFactory = tv.getRowFactory();
+        TableRow<T> tableRow = rowFactory != null ? rowFactory.call(tv) : new TableRow<>();
```

Comment:
When there is no row factory, we probably should just return like in line 632-633?

Reply:
Unlike cell factory, which always has to be present, the row factory is optional and in fact most tables will not have a row factory. If we return in that case, the algorithm will no longer work for these tables.
Compare with `javafx.scene.control.skin.TableViewSkin.createCell()`.

Reply:
You are right, thanks.